src/Files.App/Views/LayoutModes/ColumnViewBase.xaml.cs

```diff
@@ -399,20 +399,24 @@ private async void FileList_PreviewKeyDown(object sender, KeyRoutedEventArgs e)
 			}
 			else if (e.Key == VirtualKey.Left) // Left arrow: select parent folder (previous column)
 			{
-				if (!IsRenamingItem && !ParentShellPageInstance.ToolbarViewModel.IsEditModeEnabled)
-				{
-					if ((ParentShellPageInstance as ColumnShellPage)?.ColumnParams.Column > 0)
-						FocusManager.TryMoveFocus(FocusNavigationDirection.Previous);
-					e.Handled = true;
-				}
+				if (IsRenamingItem || (ParentShellPageInstance is not null && ParentShellPageInstance.ToolbarViewModel.IsEditModeEnabled) )
+					return;
+
+				var currentBladeIndex = (ParentShellPageInstance is ColumnShellPage associatedColumnShellPage) ? associatedColumnShellPage.ColumnParams.Column : 0;
+				if (currentBladeIndex == 0)
+					return;
```

Comment:
I am not sure about two things:
- Is it worth double-checking `currentBladeIndex`? (Its value is checked both here and in `MoveFocusToPreviousBlade`)
- I think `e.Handled` should be set to true anyway, as you did for `VirtualKey.Right`